client/js/main.js

```diff
@@ -429,6 +429,99 @@ async function toggleEditMode() {
   setScale();
 }
 
+// Browser support for AudioContext
+window.AudioContext = window.AudioContext || window.webkitAudioContext;
+export let context;
+export let audioBufferObj = {};
+export let sources = {};
+
+const events = ['click', 'mousedown', 'mouseup', 'dblclick', 'keydown', 'keyup', 'touchstart', 'touchend', 'focus', 'pointerdown'];
+
+events.forEach(event => {
+  document.addEventListener(event, initializeAudioContext, { once: true });
+});
+
+// Initialize AudioContext after user event
+function initializeAudioContext() {
+  if (!context && event.target.id != 'activeGameButton') {
+    context = new AudioContext();
+    audioContext(); 
+  }
+}
+
+// Function to check for routines with AUDIO function and clickSound in both state and cardDefaults
+function findAudioSources(widget) {
+  let audioSources = new Set(); // Prevents multiple audio sources for same widget
+  // Check for any routines with AUDIO function
+  Object.keys(widget.state).forEach((key) => {
+    if (key.endsWith("Routine")) {
+      const routine = widget.state[key];
+      routine.forEach((action) => {
+        if (action.func === "AUDIO") {
+          if(action.source) {
+            audioSources.add(action.source); 
+          }
+        }
+      });
+    }
+  });
```

Comment:
This construct doesn't go through subroutines like `thenRoutine` or `loopRoutine`.

Reply:
Yep, that will need to be fixed if we're keeping preloading audio.